articles/contributing/style-guide.md

```diff
@@ -0,0 +1,451 @@
+As a particular example, the singly controlled SWAP and doubly controlled NOT operations are often called the "Fredkin" and "Toffoli" operations in academic literature, but are identified in Q# primarily as `CSWAP` and `CCNOT`.
+In both cases, the API documentation comments provide synonymous names based on proper nouns, along with all appropriate citations.
+
+This preference is especially important given that some usage of proper nouns will always be necessary — Q# follows the tradition set by many classical languages, for instance, and refers to `Bool` types in reference to Boolean logic, which is in turn named in honor of George Boole.
+A few quantum concepts similarly are named in a similar fashion, including the `Pauli` type built-in to the Q# language.
+By minimizing the usage of proper nouns where such usage is not essential, we reduce the impact where proper nouns cannot be reasonably avoided.
+
+# [Guidance](#tab/guidance) 
+
+We suggest:
+
+- Avoid the use of proper nouns in names.
+
+# [Examples](#tab/examples)
+
+***
+
+### Type Conversions ###
+
+Since Q# is a strongly and staticly typed language, a value of one type can only be used as a value of another type by using an explicit call to a type conversion function.
+This is in contrast to languages which allow for values to change types implicitly (e.g.: type promotion), or through casting.
+As a result, type conversion functions play an important role in Q# library development, and comprise one of the more commonly encountered decisions about naming.
+We note, however, that since type conversions are always _deterministic_, they can be written as functions and thus fall under the advice above.
+In particular, we suggest that type conversion functions should never be named as verbs (e.g.: `ConvertToX`) or adverb prepositional phrases (`ToX`), but should be named as adjective prepositional phrases that indicate the source and destination types (`XAsY`).
+When listing array types in type conversion function names, we recommend the shorthand `Arr`.
+Barring exceptional circumstances, we recommend that all type conversion functions be named using `As` so that they can be quickly identified.
+
+# [Guidance](#tab/guidance)
+
+We suggest:
+
+- If a function converts a value of type `X` to a value of type `Y`, use either the name `AsY` or `XAsY`.
+
+# [Examples](#tab/examples)
+
+|   | Name | Description |
+|---|------|-------------|
+| ☒ | <s>`ToDouble`</s> | The preposition "to" results in a verb phrase, indicating an operation and not a function. |
+| ☒ | <s>`AsDouble`</s> | The input type is not clear from the function name. |
+| ☒ | <s>`PauliArrFromBoolArr`</s> | The input and output types appear in the wrong order. |
+| ☑ | `ResultArrAsBoolArr` | Both the input types and output types are clear. |
+
+***
+
+### Private or Internal Names ###
+
+> [!NOTE]
+> It is expected that the guidance in this section will change with Q# 1.0.
```

Comment:
Why is this here?  Can we delete this note?  What is the user supposed to do with this info?

Reply:
The intent was to provide forewarning to where breaking or incompatable changes are most likely. I'll delete, though.

Reply:
@cgranade Is there a particular reason that this conversation is still open? 
Other than that, looks good to me, so I'll approve - just make sure to address whatever still needed addressing here.

Reply:
@bettinaheim: Nope, I just forgot to hit resolve. I'll make sure that any remaining comments are captured as new Issues. Thanks!